modules/snooker/src/test/ts/browser/TableSizeTest.ts

```diff
@@ -0,0 +1,111 @@
+import { Assert, UnitTest } from '@ephox/bedrock-client';
+import { HTMLTableCellElement, HTMLTableElement } from '@ephox/dom-globals';
+import { Body, Css, Element, Insert, Remove, SelectorFind, Width } from '@ephox/sugar';
+import { ResizeDirection } from 'ephox/snooker/api/ResizeDirection';
+import { TableSize } from 'ephox/snooker/api/TableSize';
+import { Warehouse } from 'ephox/snooker/model/Warehouse';
+import * as fc from 'fast-check';
+
+const pixelTableHtml = '<table style="width: 400px"><tbody><tr><td style="width: 200px"></td><td style="width: 200px"></td></tr></tbody></table>';
+const percentTableHtml = '<table style="width: 80%"><tbody><tr><td style="width: 50%"></td><td style="width: 50%"></td></tr></tbody></table>';
+const noneTableHtml = '<table><tbody><tr><td></td><td></td></tr></tbody></table>';
+
+UnitTest.test('TableSize.getTableSize', () => {
+  const pixelTable = Element.fromHtml<HTMLTableElement>(pixelTableHtml);
+  const percentageTable = Element.fromHtml<HTMLTableElement>(percentTableHtml);
+  const noneTable = Element.fromHtml<HTMLTableElement>(noneTableHtml);
+
+  const pixelSizing = TableSize.getTableSize(pixelTable);
+  const percentageSizing = TableSize.getTableSize(percentageTable);
+  const noneSizing = TableSize.getTableSize(noneTable);
+
+  Assert.eq('Pixel sizing detected', 'pixel', pixelSizing.label);
```

Comment:
What's the behaviour if a table has a mix of percentages and pixels? What about other CSS units? Can we add some test cases for those?

Reply:
TinyMCE doesn't support that anyways and only supports the three things, so that's all that's tested here. You'll find in most places before doing operations we enforce the sizes (eg see `EnforceUnits`).